### DISCUSSION TITLE
Generate questions from text (MCQ or True False)

### DISCUSSION THREAD

**Discussion by ak4zh** [nlp-compromise/nlp_compromise#938] [General]
I am wondering if this is possible. The input will be a paragraph from a textbook or a lesson with some facts may be about some war, a programming language, human anatomy, or anything, or a block of text with some information and facts.

The output will be questions against those facts and their correct answers.

Input: 
```
Your brain has around 100 billion nerve cells. 

An adult brain weighs three pounds.

The brain cannot feel pain.

Did you know the technique of fingerprinting is known as dactyloscopy?
```

Output:
```
How many nerve cells your brain has? 
How much does an adult brain weigh?
Can the brain feel the pain? 
What is the technique of fingerprinting called?
```

The text will obviously not be just a series of one-line facts it can be a long paragraph, so we need to extract the sentences that have some information and convert them to questions with correct and may be incorrect answers as well to make an MCQ

**Comment by TSavo** [comment]
I don’t think Compromise does this, but it’s the type of thing GPT-3 is
extremely well suited to do. Just give it a prompt of example answers, and
their questions, then give it the answer you want a question for, and
prompt it for the question.

Sent via Superhuman iOS ***@***.***>

**Comment by spencermountain** [comment]
hey @Ak4zh - yeah, if you knew what sort of facts you were expecting in the text, you could hunt+peck for them, like this
https://runkit.com/spencermountain/62d2b7c631c0e70008185a59

that would let you pull-out the easiest fact-like details, that you see. It would be pretty fun to do.
you are limited in having to create the templates yourself, like @TSavo said, a vector-based system might scoop-up harder wordings and formats
cheers
